src/components/common/FilterMultiSelect/FilterMultiSelect.test.tsx

```diff
@@ -0,0 +1,111 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import React from 'react';
+import FilterMultiSelect from '.';
+
+test('Apply calls onConfirm with selection', () => {
+  const onCancel = jest.fn();
+  const onConfirm = jest.fn();
+  render(
+    <FilterMultiSelect
+      label='Test'
+      initialSelection={[1, 2]}
+      onCancel={onCancel}
+      onConfirm={onConfirm}
+      options={[1, 2, 3]}
+      renderOption={(id) => `${id}`}
+    />
+  );
+
+  // should see only the option 1 and 3 pills
+  expect(screen.queryByText('1')).toBeInTheDocument();
+  expect(screen.queryByText('2')).not.toBeInTheDocument();
+  expect(screen.queryByText('3')).toBeInTheDocument();
+
+  // remove pill 1
+  const removeButtons = screen.getAllByLabelText('remove');
+  expect(removeButtons.length).toBe(2);
+  fireEvent.click(removeButtons[0]);
+
+  // should see only the option 3 pill
+  expect(screen.queryByText('1')).not.toBeInTheDocument();
+  expect(screen.queryByText('2')).not.toBeInTheDocument();
+  expect(screen.queryByText('3')).toBeInTheDocument();
+
+  // click the apply button
+  const applyButton = screen.getByText('Apply');
```

Comment:
some suggestions for future consideration:
get by id or class - it's more specific in what we are looking for.
same to check for pills, get by class, check count of elements returned.. check html value of elements returned.

Reply:
Thanks. I can make those improvements 😊 

Right now having some trouble getting this to run, I committed this by mistake.